Datdy/Feat/entries

ThanhDatVu111 opened the pull request:
## 🚀 What does this PR do?

> This PR implements full “entry” support inside a specific book. In other words, once you open a book, you can now create, store, and retrieve rich “entries” (memories) tied to that book. The major pieces are:


<img width="1158" alt="Screenshot 2025-05-31 at 10 49 30 PM" src="https://github.com/user-attachments/assets/f4c0ade0-68e0-4307-ae19-06fa84b62638" />

---

## ✅ Checklist

- [x] Code compiles & runs
- [ ] Lint checks pass
- [ ] Updated `.env.example` if needed
- [ ] PR linked to related issue/ticket (if applicable)

---

## 🔍 How to test this

**Empty entry screen**

<img width="499" alt="Screenshot 2025-05-31 at 10 34 08 PM" src="https://github.com/user-attachments/assets/37a05b49-2483-41e2-9908-970ac4ce105b" />

**Entry screen**

<img width="499" alt="Screenshot 2025-05-31 at 10 30 29 PM" src="https://github.com/user-attachments/assets/1a05cb98-30ff-4cbf-b593-027c821b2e17" />

**Backend**

<img width="1321" alt="Screenshot 2025-05-31 at 10 30 04 PM" src="https://github.com/user-attachments/assets/0db7a65d-fe2f-45ad-83c6-56398a04b4e0" />

---

## 🧠 Extra Notes

> Tommorrow I will focus on how to display the entry

ThanhDatVu111 committed:
feat(backend): Add entries model, service, and controller for entry creation and retrieval

- Implemented  function in  to insert new entries into the database.
- Added  function in  to fetch entries by .
- Created  and  functions in  for business logic.
- Developed  and  controllers in  to handle API requests.
- Added routes in  for entry creation () and retrieval ().

tranguv commented on backend/src/controllers/entriesController.ts:
```diff
@@ -0,0 +1,50 @@
+  } catch (err: any) {
+    next(err);
+  }
+}
```
Will delete and update endpoints are in another PR or hmm

ThanhDatVu111 replied:
delete and edit are in the latest commit

ThanhDatVu111 committed:
handle entry updates but still has bugs
- Time is not updated correctly in the UI after editing an entry.
- Unable to remove previously added images when editing an entry.

ThanhDatVu111 committed:
add API endpoints for updating and deleting entries
- Added PUT endpoint for updating entries.
- Added DELETE endpoint for removing entries.
- Improved error handling for entry-related operations.

ThanhDatVu111 committed:
minor changes

ThanhDatVu111 committed:
fix location little bit

ThanhDatVu111 committed:
resolve merge conflict

ThanhDatVu111 committed:
make sure the [bookId] work with Quyen Library

ThanhDatVu111 committed:
minor changes before checkin for library/entry feat

ThanhDatVu111 committed:
fix the photo galary, image fetching and location log

ThanhDatVu111 committed:
buggggg

ThanhDatVu111 committed:
bugggg in backend

ThanhDatVu111 committed:
add cloudinary

ThanhDatVu111 committed:
I change the library logic to make it less complicated

ThanhDatVu111 committed:
Merge branch 'main' of https://github.com/ThanhDatVu111/0-KM into feat/entries

ThanhDatVu111 committed:
change book UI and also add cloudinary in entry

ThanhDatVu111 committed:
add cloudinary in server

ThanhDatVu111 committed:
refractor the whole entry creation

ThanhDatVu111 committed:
remove cloudinary file

ThanhDatVu111 committed:
fix update api call to make sure it work with cloudinary

ThanhDatVu111 committed:
remove multer

ThanhDatVu111 committed:
done with entry

ThanhDatVu111 committed:
actually I'm done with entries

ThanhDatVu111 committed:
expand entry

ThanhDatVu111 committed:
add leaflet open source for mapping intergration

ThanhDatVu111 committed:
just some new packages

ThanhDatVu111 committed:
enable clear button and also put in explainatoin for others

ThanhDatVu111 committed:
fix entries and cloudinary fetching

ThanhDatVu111 committed:
add socket io for editing together

ThanhDatVu111 committed:
add backend for socket.io

ThanhDatVu111 committed:
add initialbody and initialtitle into ydoc for edit text

ThanhDatVu111 committed:
resolve merge conflict and add new commit to match main

ThanhDatVu111 committed:
minor fixed

ThanhDatVu111 committed:
enable video

ThanhDatVu111 committed:
resolve merge conflict

ThanhDatVu111 committed:
resolve local conflict

ThanhDatVu111 committed:
fixed UI and entry bug

ThanhDatVu111 committed:
resolve merge conflicts

ThanhDatVu111 committed:
minor change

ThanhDatVu111 committed:
testing

ThanhDatVu111 committed:
add realtime for pairing

ThanhDatVu111 committed:
add package lock

ThanhDatVu111 committed:
update pairroom

ThanhDatVu111 committed:
add realtime and also fix little bit for entries
